theano/gof/cmodule.py

```diff
@@ -2310,5 +2319,314 @@ def print_command_line_error():
+
+                lines = BytesIO(stdout + stderr).readlines()
+                lines = decode_iter(lines)
+                if parse:
+                    selected_lines = []
+                    for line in lines:
+                        if "AVX" in line:
+                            selected_lines.append(line.strip())
+                        elif "SSE" in line:
+                            selected_lines.append(line.strip())
+                    lines = list(set(selected_lines))  # to remove duplicate
+
+                return lines
+
+            # The '-' at the end is needed. Otherwise, g++ do not output
+            # enough information.
+            native_lines = get_lines("%s -xHOST -E -v -" % theano.config.cxx)
+            if native_lines is None:
+                _logger.warn("Call to 'icpc -xHOST' failed,"
+                             "not setting -xHOST flag")
+                detect_march = False
+            else:
+                _logger.info("icpc -xHOST selected lines: %s",
+                             native_lines)
+
+        # Add the -xHOST
+        cxxflags.extend(["-xHOST"])
+
+        # NumPy 1.7 Deprecate the old API. I updated most of the places
+        # to use the new API, but not everywhere. When finished, enable
+        # the following macro to assert that we don't bring new code
+        # that use the old API.
+        cxxflags.append("-DNPY_NO_DEPRECATED_API=NPY_1_7_API_VERSION")
+        numpy_ver = [int(n) for n in numpy.__version__.split('.')[:2]]
+
+        # numpy 1.7 deprecated the following macro but the new one didn't
+        # existed in the past
+        if bool(numpy_ver < [1, 7]):
+            cxxflags.append("-DNPY_ARRAY_ENSUREARRAY=NPY_ENSUREARRAY")
+            cxxflags.append("-DNPY_ARRAY_ENSURECOPY=NPY_ENSURECOPY")
+            cxxflags.append("-DNPY_ARRAY_ALIGNED=NPY_ALIGNED")
+            cxxflags.append("-DNPY_ARRAY_WRITEABLE=NPY_WRITEABLE")
+            cxxflags.append("-DNPY_ARRAY_UPDATE_ALL=NPY_UPDATE_ALL")
+            cxxflags.append("-DNPY_ARRAY_C_CONTIGUOUS=NPY_C_CONTIGUOUS")
+            cxxflags.append("-DNPY_ARRAY_F_CONTIGUOUS=NPY_F_CONTIGUOUS")
+
+        # Platform-specific flags.
+        # We put them here, rather than in compile_str(), so they en up
+        # in the key of the compiled module, avoiding potential conflicts.
+
+    
+        # Figure out whether the current Python executable is 32
+        # or 64 bit and compile accordingly. This step is ignored for
+        # ARM (32-bit and 64-bit) architectures in order to make
+        # Theano compatible with the Raspberry Pi, Raspberry Pi 2, or
+        # other systems with ARM processors.
+        """
+        if (not any(['arm' in flag for flag in cxxflags]) and
+                not any(arch in platform.machine() for arch in ['arm', 'aarch'])):
+            n_bits = local_bitwidth()
+            cxxflags.append('-m%d' % n_bits)
+            _logger.debug("Compiling for %s bit architecture", n_bits)
+        """
```

Comment:
If icpc don't support arm, just remove all that code and the corresponding comment.